ProcessMaker/modeler
# PM Blocks Menu Implementation (#FOUR-7370)

This pull request introduces the PM Blocks menu feature, addressing ticket [FOUR-7370](https://processmaker.atlassian.net/browse/FOUR-7370). The purpose of this feature is to enhance the Modeler tab by incorporating the display and utilization of PM Blocks. PM Blocks can now be seamlessly integrated within the modeler and executed as part of a process.

## Changes:

- Introduces a new method, `generateExports`, responsible for initiating an API call to the database. This call retrieves the stored PM Blocks and processes the response. The method iterates over the retrieved data and configures the corresponding node object. The resulting PM Blocks are then exported and stored in an array.

- Implements a new method called `registerPmBlocks`, which ensures the proper registration of PM Blocks within the modeler. This method is triggered during the initialization of the modeler, enabling seamless integration of PM Blocks functionality.

## To Test:

1. Ensure that the [PM Blocks](https://github.com/ProcessMaker/package-pm-blocks/pull/3) package branch  is `feature/FOUR-7370` installed.
2. Verify that PM Blocks are properly configured in the application.
3. Create a new process within the Modeler.
4. Check that the PM Block tab is visible in the Modeler interface.
5. Click on the PM Block tab to ensure it opens successfully and displays the list of available PM Blocks.
6. Click and drag a PM Block from the PM Block tab onto the modeler BPMN canvas.
7. Confirm that the dropped PM Block is visually represented on the canvas and can be interacted with (e.g., moving, resizing, connecting with other elements).
8. Run the process that includes the PM Block.
9. Validate that the process executes the associated PM Block properly when triggered during the process flow.
10. Verify that the PM Block functions as expected, performing the intended actions within the process.
11. Perform additional tests with different PM Blocks, covering various scenarios and functionalities.


## Related Tickets & Packages
-  PM Block PR - https://github.com/ProcessMaker/package-pm-blocks/pull/3
- Ticket: [FOUR-7370](https://processmaker.atlassian.net/browse/FOUR-7370)

## Code Review Checklist


[FOUR-7370]: https://processmaker.atlassian.net/browse/FOUR-7370?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FOUR-7370]: https://processmaker.atlassian.net/browse/FOUR-7370?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
